Creating a virtual assistant using python (#2008)
section-engineering-education/engineering-education: articles/creating-a-virtual-assistant-using-python/index.md

---
layout: engineering-education
status: publish
published: true
url: /engineering-education/creating-a-virtual-assistant-using-python/
title: Creating a Virtual Assistant Using Python
description: This article takes the reader through creating a virtual assistant using python. A virtual assistant, also called an AI assistant or digital assistant, is an application program that understands natural language voice commands and completes tasks for the user.
author: aransiola-ayodele
date: 2021-04-12T00:00:00-01:00
topics: []
excerpt_separator: <!--more-->
images:
  - url: /engineering-education/creating-a-virtual-assistant-using-python/hero.jpg
    alt: Virtual Assistant Image Example
---

> A virtual assistant, also known as an AI assistant or a digital assistant, is a computer program that recognizes natural language voice commands and performs tasks for the user. This article will use a python library to incorporate Alexa and will guide you through the process of creating your own voice assistant.

Reviewer: ```suggestion
A virtual assistant, also known as an AI assistant or a digital assistant, is a computer program that recognizes natural language voice commands and performs tasks for the user. This article guides you through the process of creating your own voice assistant through the use of a python library. 
```

2. Python Text To Speech v3(Pyttsx3): this library is a text-to-speech conversion library in Python. It works offline and is compatible with both Python 2 and 3.

This package can be downloaded [here](https://pypi.org/project/pyttsx3/). You will copy the command found on the website and paste in the terminal just as you have done to install the first library.
![Python text to speech library download page](/engineering-education/creating-a-virtual-assistant-using-python/pyttsx3.png)

Once you have successfully installed the second package, we will continue to install the third package called:

3. Python Audio (PyAudio): this library provides Python bindings for PortAudio, the cross-platform audio I/O library. PyAudio allows you to quickly play and record audio on a number of devices using Python. Download the package [here](https://pypi.org/project/PyAudio/). Install the package using the steps used in 1 above.
   ![Python Audio library download page](/engineering-education/creating-a-virtual-assistant-using-python/pyaudio.png)

### Note:

> for Linux users, installing the Pyaudio, you might encounter an error, to fix the error and install PyAudio successfully, kindly follow the steps below:

Reviewer: ```suggestion
For Linux users, you might encounter an error when installing Pyaudio. Follow the following steps to fix the error:
```

1. Install **_portaudio_** modules using this command:

```bash
sudo apt-get install libasound-dev
```

2. Download the portaudio archive from: [Link](http://files.portaudio.com/download.html)

3. Unzip the archive using:

```terminal
tar -zxvf [filename portaudio.tgz]
```

4. Enter the directory of the unzipped portaudio file, then run:

```terminal
 ./configure && make
```

5. Once that is successful, Install it by using this command:

```terminal
sudo make install
```

6. Lastly, install pyaudio:

```terminal
pip install pyaudio
```

> We have successfully installed the three libraries. Now, let get our Virtual assistant alive!!!

Reviewer: ```suggestion
We have successfully installed the three libraries. Now, let's get our Virtual assistant alive!!!
```
